File: generator/src/test/resources/kotlin/expected/sunday/RequestMethodsTest/test-request-method-generation-with-result-response.output.kt
package io.test

import io.outfoxx.sunday.MediaType
import io.outfoxx.sunday.RequestFactory
import io.outfoxx.sunday.http.Method
import io.outfoxx.sunday.http.Request
import io.outfoxx.sunday.http.Response
import io.outfoxx.sunday.http.ResultResponse
import kotlin.Unit
import kotlin.collections.List

public class API(
  public val requestFactory: RequestFactory,
  public val defaultContentTypes: List<MediaType> = listOf(MediaType.JSON),
  public val defaultAcceptTypes: List<MediaType> = listOf(MediaType.JSON),
) {
  public suspend fun fetchTest(): ResultResponse<Test> = this.requestFactory
    .resultResponse(
      method = Method.Get,
      pathTemplate = "/tests",
      acceptTypes = this.defaultAcceptTypes
    )

  public suspend fun putTest(body: Test): ResultResponse<Test> = this.requestFactory
    .resultResponse(
      method = Method.Put,
      pathTemplate = "/tests",
      body = body,
      contentTypes = this.defaultContentTypes,
      acceptTypes = this.defaultAcceptTypes
    )

  public suspend fun postTest(body: Test): ResultResponse<Test> = this.requestFactory
    .resultResponse(
      method = Method.Post,
      pathTemplate = "/tests",
      body = body,
      contentTypes = this.defaultContentTypes,
      acceptTypes = this.defaultAcceptTypes
    )

  public suspend fun patchTest(body: Test): ResultResponse<Test> = this.requestFactory
    .resultResponse(
      method = Method.Patch,
      pathTemplate = "/tests",
      body = body,
      contentTypes = this.defaultContentTypes,
      acceptTypes = this.defaultAcceptTypes
    )

  public suspend fun deleteTest(): ResultResponse<Unit> = this.requestFactory
    .resultResponse(
      method = Method.Delete,
      pathTemplate = "/tests"
    )

  public suspend fun headTest(): ResultResponse<Unit> = this.requestFactory
    .resultResponse(
      method = Method.Head,
      pathTemplate = "/tests"
    )

  public suspend fun optionsTest(): ResultResponse<Unit> = this.requestFactory
    .resultResponse(
      method = Method.Options,
      pathTemplate = "/tests"
    )

  public suspend fun patchableTest(body: PatchableTest): ResultResponse<Test> = this.requestFactory
    .resultResponse(
      method = Method.Patch,
      pathTemplate = "/tests2",
      body = body,
      contentTypes = this.defaultContentTypes,
      acceptTypes = this.defaultAcceptTypes
    )

  public suspend fun requestTest(): Request = this.requestFactory
    .request(
      method = Method.Get,
      pathTemplate = "/request",
      acceptTypes = this.defaultAcceptTypes
    )

  public suspend fun responseTest(): Response = this.requestFactory
    .response(
      method = Method.Get,
      pathTemplate = "/response",
      acceptTypes = this.defaultAcceptTypes
    )
}

File: generator/src/test/resources/swift/expected/RequestMethodsTest/test-request-method-generation-with-result-response.output.swift
import Foundation
import Sunday

public class API {

  public let requestFactory: RequestFactory
  public let defaultContentTypes: [MediaType]
  public let defaultAcceptTypes: [MediaType]

  public init(
    requestFactory: RequestFactory,
    defaultContentTypes: [MediaType] = [.json],
    defaultAcceptTypes: [MediaType] = [.json]
  ) {
    self.requestFactory = requestFactory
    self.defaultContentTypes = defaultContentTypes
    self.defaultAcceptTypes = defaultAcceptTypes
  }

  public func fetchTest() async throws -> ResultResponse<Test> {
    return try await self.requestFactory.resultResponse(
      method: .get,
      pathTemplate: "/tests",
      pathParameters: nil,
      queryParameters: nil,
      body: Empty.none,
      contentTypes: nil,
      acceptTypes: self.defaultAcceptTypes,
      headers: nil
    )
  }

  public func putTest(body: Test) async throws -> ResultResponse<Test> {
    return try await self.requestFactory.resultResponse(
      method: .put,
      pathTemplate: "/tests",
      pathParameters: nil,
      queryParameters: nil,
      body: body,
      contentTypes: self.defaultContentTypes,
      acceptTypes: self.defaultAcceptTypes,
      headers: nil
    )
  }

  public func postTest(body: Test) async throws -> ResultResponse<Test> {
    return try await self.requestFactory.resultResponse(
      method: .post,
      pathTemplate: "/tests",
      pathParameters: nil,
      queryParameters: nil,
      body: body,
      contentTypes: self.defaultContentTypes,
      acceptTypes: self.defaultAcceptTypes,
      headers: nil
    )
  }

  public func patchTest(body: Test) async throws -> ResultResponse<Test> {
    return try await self.requestFactory.resultResponse(
      method: .patch,
      pathTemplate: "/tests",
      pathParameters: nil,
      queryParameters: nil,
      body: body,
      contentTypes: self.defaultContentTypes,
      acceptTypes: self.defaultAcceptTypes,
      headers: nil
    )
  }

  public func deleteTest() async throws -> ResultResponse<Void> {
    return try await self.requestFactory.resultResponse(
      method: .delete,
      pathTemplate: "/tests",
      pathParameters: nil,
      queryParameters: nil,
      body: Empty.none,
      contentTypes: nil,
      acceptTypes: self.defaultAcceptTypes,
      headers: nil
    )
  }

  public func headTest() async throws -> ResultResponse<Void> {
    return try await self.requestFactory.resultResponse(
      method: .head,
      pathTemplate: "/tests",
      pathParameters: nil,
      queryParameters: nil,
      body: Empty.none,
      contentTypes: nil,
      acceptTypes: self.defaultAcceptTypes,
      headers: nil
    )
  }

  public func optionsTest() async throws -> ResultResponse<Void> {
    return try await self.requestFactory.resultResponse(
      method: .options,
      pathTemplate: "/tests",
      pathParameters: nil,
      queryParameters: nil,
      body: Empty.none,
      contentTypes: nil,
      acceptTypes: self.defaultAcceptTypes,
      headers: nil
    )
  }

  public func patchableTest(body: PatchableTest) async throws -> ResultResponse<Test> {
    return try await self.requestFactory.resultResponse(
      method: .patch,
      pathTemplate: "/tests2",
      pathParameters: nil,
      queryParameters: nil,
      body: body,
      contentTypes: self.defaultContentTypes,
      acceptTypes: self.defaultAcceptTypes,
      headers: nil
    )
  }

  public func requestTest() async throws -> URLRequest {
    return try await self.requestFactory.request(
      method: .get,
      pathTemplate: "/request",
      pathParameters: nil,
      queryParameters: nil,
      body: Empty.none,
      contentTypes: nil,
      acceptTypes: self.defaultAcceptTypes,
      headers: nil
    )
  }

  public func responseTest() async throws -> (Data?, HTTPURLResponse) {
    return try await self.requestFactory.response(
      method: .get,
      pathTemplate: "/response",
      pathParameters: nil,
      queryParameters: nil,
      body: Empty.none,
      contentTypes: nil,
      acceptTypes: self.defaultAcceptTypes,
      headers: nil
    )
  }

}
